### ISSUE TITLE
Home: Misplaced Main Search Submit Button

### ISSUE THREAD

**Issue by jscholes** [benetech/ImageShare#55]
The button to submit the search needs to appear after the relevant selects below the input, not before.  Users, particularly those operating a screen reader or magnification, will just assume that the search button marks the end of the form and not move any further, resulting in them missing the selects altogether.

**Comment by clapierre** [comment]
@jscholes Not sure which image you are referring to.  
Is this the Magnifying glass at the top of the Landing page?  Or are you referring to the Search button at the bottom.  Attached are the two images I am referring to from the mockups.
![Screen Shot 2020-04-22 at 12 33 41 PM](https://user-images.githubusercontent.com/7736749/80025573-b9c65d80-8495-11ea-82cc-e79895c83d1b.png)
![Screen Shot 2020-04-22 at 12 34 22 PM](https://user-images.githubusercontent.com/7736749/80025585-bc28b780-8495-11ea-8114-b1c2df8df081.png)

**Comment by clapierre** [comment]
If we are talking about the 2nd image the Search button is after the Search field where it has: Enter name of collection or resource to search (try "geology" or "heart".   So not sure where you were expecting to see the Search button as it is after the search field.

The other screenshot above (ie. first magnifying glass image beside the text "Search Resources" is just a quick link down to the search input form.

Our designer thought we needed a quick way to bypass the collections and jump down to the search input field.

If you feel we don't need that we can always remove it, but I think our designer was worried about a smaller screen and having to scroll down a ways to get to the search field.

**Comment by sinabahram** [comment]
I’m copying @vincent-young-1701 into this in case he can also help clarify.

**Comment by sinabahram** [comment]
I’m not a fan of a skip link to a specific component. This presents all sorts of issues. If we think search belongs somewhere, that’s where it should go, but not a nav link down, you know?

**Comment by jscholes** [comment]
@clapierre

![The main search button](https://user-images.githubusercontent.com/6454755/80026292-85f43380-84a7-11ea-8746-5f213a24bbe6.png)

**Comment by clapierre** [comment]
@Sina, Yeah I get it, maybe that could be handled with a menu item to "Search Resources" that will jump down to the component in question on the landing page.  Our designer was worried they may only think you can search via collections, and if we move the Search up higher than the collections, one might not know there are collections to pick from.  Maybe its a mute point and folks will figure out they need to scroll down a little further if on a smaller screen and the search functionality is "below the fold" as they would call it back in the day.

**Comment by clapierre** [comment]
@jscholes Ok I see you are referring to the 2nd instance of the Search Button so what is the problem?
Are you concerned that the refinements to the search (ie. Subject, Resource Type and Accessibility accommodation) appears after the search button itself, and those should be brought up above the Search Input Field and button?

**Comment by vincent-young-1701** [comment]
@clapierre I'm going to take a look at this today and let you know what I come up with.  I don't think there will be a "below the fold" issue on the home if you were to do that.  You can compress the vertical height of the search component as well as the header.

**Comment by vincent-young-1701** [comment]
I also think you could have the same interaction with this search as you have on the inner pages where the "Advanced" is collapsed and users decide to expand.  Anyway, there are lots of options to ensure the collections are revealed to users.

**Comment by vincent-young-1701** [comment]
@clapierre this is an issue on both of the search forms. I'd do one of two things:

1. Have the search input span the width of it's container, following by the refinement controls on another row, then on another row have the search button right or left aligned.

2. On the same row the search input > advanced search button (expanding/collapsing) > Search Button.  Expanding the refinements would put them underneath on a new row.

Personally, I'd go with the second choice.  It cleans up the interface and removes tab stops for users not interested.  If expanded, users, would tab into the refinement controls and the last refinement control would precede the search button in the tab order so screen reader users would not miss that.

**Comment by clapierre** [comment]
@vincent-young-1701 Would you be able to mock up that 2nd option?  I just would like to see it first.

**Comment by jscholes** [comment]
@vincent-young-1701 

> On the same row the search input > advanced search button (expanding/collapsing) > Search Button. Expanding the refinements would put them underneath on a new row.

If the search button is on the row above the expanded refinement dropdowns, but the refinement controls appear in the tab order before the search button, that would create a confusing tab order, no?  Your focus would move down a row, across the second row, then back up to the search button in the row above.

@clapierre 

> Are you concerned that the refinements to the search (ie. Subject, Resource Type and Accessibility accommodation) appears after the search button itself, and those should be brought up above the Search Input Field and button?

In part.  The concern is that they are after the search button, but having them after the input is fine (and probably makes the most sense).

**Comment by clapierre** [comment]
Right I am fine moving the Search button down if required, seems a little odd  but maybe not, similar I guess here in Github where you have this text input filed, then a section to add attachments, then 2 buttons Close and comment and the big Green Comment button.  

But here the button is right aligned to this input text area.  Where you were concerned about right aligning right vs. aligning to the left?  Is this something of a point of contention between you all and Github as well? :). Or is this different enough and people expect that a "submit" type button will be right alined with an input text area if not on the same line as the field.

**Comment by jscholes** [comment]
@clapierre Believe me, I fully intend to battle to the bitter end to get some of these GitHub buttons reordered...

**Comment by sinabahram** [comment]
Just a note, please let’s not use GitHub as any standard of a11y. None of our design feedback has been incorporated what-so-ever, full stop.

**Comment by sinabahram** [comment]
Charles, I think these are good points you and the designer are raising; however, I also think Vincent’s updates to designs may address all of them in one unified way.

**Comment by clapierre** [comment]
Understood @sinabahram!  That sounds good Sina, we can table most of these presumably until after that has been finished and reviewed.

**Comment by vincent-young-1701** [comment]
@jscholes Not for this interaction...yes, you're right in that it is not perfect, but no implementation is going to give everything everyone wants.  It would be collapsed, so the user would already identify the intention when activating and the next logical place after the last refinement would be the search input.  Doing it this way is going to get rid of tab stops people are not looking for and bring up the collections closer to the top of the page so they are not missed on smaller screens.  @clapierre check out the designs below, but please keep in mind I cooked these up very quickly and they are more mocks than anything else.  What we've got in the first image is the "Search" heading then the text, "Library of 5,074 accessible STEM media resources", below that is label "Search collection or resource", and below is the input with placeholder, Try “geology” or “heart”, right aligned is the "Advanced Search" control, and right aligned to that is the "Search Library" button.  The second image has the advanced search expanded with the three example selects displayed.  Let me know what you guys think.  This mock would show the design a bit more modern/clean.
![collapsed](https://user-images.githubusercontent.com/25315048/80037365-4f6ae880-84a8-11ea-9188-b7dba1258f32.png)
![expanded](https://user-images.githubusercontent.com/25315048/80037374-5265d900-84a8-11ea-8a3d-996292dab803.png)

**Comment by clapierre** [comment]
This looks fine to me, I assume the Advance Search carrot in the first image would be pointing to the Right not down like it is in the 2nd instance when the advance search options are showing correct?  Just like how the HTML Summary/details carrot looks collapsed vs. expanded.

**Comment by vincent-young-1701** [comment]
Yup, right on.  We can have unification there across the site for those aided interactions.

**Comment by vincent-young-1701** [comment]
Actually... I was thinking it would point down when collapsed, and point up when expanded, but we can iron that out.

**Comment by jscholes** [comment]
@vincent-young-1701 Sorry, I really don't agree, on two counts:

1. I don't know if the last refinement select should come before the search input.  Until I've typed a search string, or at least been made aware that I can, there's nothing for me to refine.  No context for refinement.
2. Regardless of expanding/collapsing behaviour, a visually illogical tab order can't be excused for interactive controls.  It needs to follow the visual order, full-stop.  We can have a disclosure for the dropdowns by all means, but its presence doesn't let us off the hook when it comes to focus ordering.

To sum up, I think the collapsed order should be:

1. Search input
2. Advanced search disclosure
3. Search button

And the expanded order:

1. Search input
2. Advanced search disclosure
3. First dropdown
4. Second Dropdown
5. Third dropdown
6. Search button

You can style this however you like, as long as the DOM and visual order match.

**Comment by vincent-young-1701** [comment]
Yup james this is exactly spot on.  I like the idea of search disclosure directly before the input semantically and visually. @clapierre does that make sense?  Instead of the advanced search dropdown after the input you’d have it before.  As james said this is good for screen reader users and all as it will save everyone from potential unnecessary typing.  I’ll mock something up, placing it before in the sane row or directly above the input.

**Comment by jscholes** [comment]
> I like the idea of search disclosure directly before the input semantically and visually.

That's not what I intended to imply.  The disclosure, and the dropdowns that it controls, should be between the search input and the search submit button.

**Comment by jscholes** [comment]
If you have the disclosure before the input, but the dropdowns after the input, the disclosure doesn't appear to have an immediate impact for a screen reader user.  Because the button is separated from the controlled elements.

**Comment by vincent-young-1701** [comment]
Sorry I misread the intention.  What you’ve described is what I had originally put down for the interaction.

**Comment by sinabahram** [comment]
I completely agree with @jscholes on this. Just want to make sure we’re all on the same page.

**Comment by vincent-young-1701** [comment]
So the advanced search if placed before the search input would expand the drop downs before the input.  I’m happy to do either way, Advanced search before or after the input.  They both have validity in my opinion.

**Comment by sinabahram** [comment]
I think edit first, then advanced search, collapsed, then search button. This will reflect the logical ordering most folks expect.

**Comment by clapierre** [comment]
So can we get an updated view of what this looks like because it sounds like we need to visually/physically move the Search Button down below the search Field  right?  That will further move down if the advance search options are opened.  
Did I understand this correctly?

**Comment by vincent-young-1701** [comment]
Sure, let me mock that up in a minute.

**Comment by vincent-young-1701** [comment]
The search button here is left aligned under the input.  I wouldn't suggest a full width, unless you are on mobile.  The placeholder, Try searching “geology” or “heart”, is placed under the input so it is visible at all times.  Disclosure carrot points down when collapsed and vertically  when expanded. 
  
![collapsed](https://user-images.githubusercontent.com/25315048/80047879-72ef5c80-84c3-11ea-93f2-ad9387c62bd4.png)
![expanded](https://user-images.githubusercontent.com/25315048/80047886-78e53d80-84c3-11ea-819b-b7f4f5271c4c.png)

**Comment by vincent-young-1701** [comment]
Sorry about all the back and forth, but getting this right is important as it's one of the main interactive features.  I've come up with something that I think is going to work. @clapierre I didn't like the Submit button being vertically aligned as opposed to horizontal to the input because users could possibly miss the Advanced Search button all the way to the right of the input.  So, instead the search button is back to the right of the Advanced Search and when expanded the submit button moves down with the expansion.   Don't worry about any of the copy, just placement and interaction for this mock.  @jscholes @sinabahram how you feeling about this approach?  
![collapsed](https://user-images.githubusercontent.com/25315048/80058787-1a2dbd00-84df-11ea-82c4-3b303a9d5bd8.png)
![expanded](https://user-images.githubusercontent.com/25315048/80058829-2f0a5080-84df-11ea-8da5-a098b3a67a26.png)

**Comment by clapierre** [comment]
Interesting approach, I was also wondering about that myself, I am not sure I have see a button move like that before.

Just wondering if instead you have the Search Field the entire width 
Then have Advance search right under it that can be expanded, and then under that have the Search button that will move further down if advance is expanded.

Also why have you moved "Try Searching for 'geology' or 'heart' from inside the field as a hint and have it as text below the search field.  WOuldn't that mean AT user's wouldn't encounter that unless they skip past the search field?

**Comment by jscholes** [comment]
@clapierre 

> Also why have you moved "Try Searching for 'geology' or 'heart' from inside the field as a hint and have it as text below the search field.

The original placement implied that it was a placeholder.  Placeholders are not accessible - see https://github.com/benetech/Imageshare/issues/54.

**Comment by clapierre** [comment]
Yup ok makes sense.

**Comment by vincent-young-1701** [comment]
@clapierre yes if we have more vertical space to work with that’s doable; I thought we were trying to keep the module tight to ensure the collections are more in view.  FYI, I did try a full width submit button and it seemed out of place on a 1200px width...on mobile I have that mocked up and full width works great on those smaller viewports.

**Comment by clapierre** [comment]
Right, I wasn't saying the submit button needs to be full width, that would be the search input field.  the submit button under it can look like what you have above in the first two examples or whatever makes sense to fit the text of the button.

**Comment by vincent-young-1701** [comment]
Meant to say full width controls.  Sure I’ll create another mockup.   If vertical height is no issue, this is the route we should be going, input, expand/collapse, submit, vertically stacked.

**Comment by clapierre** [comment]
I think that makes the most sense, and I would think would be easier to code than having to move the Submit button down if the advance details is expanded.  This way HTML should do all of that for us with the Advance Search being an HTML Summary/Details control, and the Search Button showing up afterwards.  

Don't get me wrong I like the Submit button on the right of the search field/advance search control but am just worried if extra coding will have to occur to make it behave like you have in the mock-ups, and if you have ever seen a site that does this and is accessible and the visual matches the tab order with expanded/collapsed.

**Comment by vincent-young-1701** [comment]
Sure, completely understand.  There is something I want to make sure I understand before I do more mocking.  If the advanced controls are expanded by default, will this possibly cause confusion for users thinking the refinements are necessary in order to perform a search?  I would think if you want them expanded by default, a heading would be necessary to let them know they are for...Advanced Search.  This is what the expanding/collapsing was solving in my mind, making that functionality explicit for those that want to use it.

**Comment by clapierre** [comment]
Yes, I think most users won't need that advance searching so hiding it until they specifically want it makes sense to me which is why we thought it would be collapsed initially.  Same thing on the Search results page for its Advance Search options.

**Comment by vincent-young-1701** [comment]
OK then, I think this mock will work with the advanced search collapsed by default under the input and search button below all that.  Thoughts?
![collapsed](https://user-images.githubusercontent.com/25315048/80145441-07f66200-8565-11ea-9f28-6a37b4f3382e.png)
![expanded](https://user-images.githubusercontent.com/25315048/80145476-13498d80-8565-11ea-92b6-ef13272b6231.png)

**Comment by sinabahram** [comment]
100% agreement on hiding advanced by default.

**Comment by clapierre** [comment]
This looks good, lets just not spend a lot of time with how the HTML Summary/details expand/collapse caret is presented lets see what we get out of the box with standard HTML controls and if we need then we can customize the look and feel if it doesn't add a lot of development time.  Just don't want spending a few hours to get how the caret looks just right.  I think using default HTML elements will make for quick and easy development and maintenance.

**Comment by sinabahram** [comment]
Charles, totally agree on native markup wherever possible.

**Comment by vincent-young-1701** [comment]
Should be able to use some UTF characters.

**Comment by vincent-young-1701** [comment]
@clapierre FYI, as you get down to mobile, the layout doesn't really change and is just compressed.

![Mobile-collapsed](https://user-images.githubusercontent.com/25315048/80161287-bcec4700-8584-11ea-8f37-05dc230baae7.png)
![Mobile-expanded](https://user-images.githubusercontent.com/25315048/80161291-bfe73780-8584-11ea-9211-5c2e69a5f76a.png)

**Comment by clapierre** [comment]
Looks good!

**Comment by clapierre** [comment]
@vincent-young-1701 Why did we need to reopen this ticket?

**Comment by vincent-young-1701** [comment]
Sorry Charles, this was an accident!  I accidentally hit open and thought I hit close, but I guess the close event didn't propagate.  Closing now.
